### PR TITLE
Remove async in sending instructions, refactor send method

### PR DESCRIPTION
**Pull request by mirpo**

### CODE REVIEW

File: src/connection/WsEndpoint.js [hunk 1]
```diff
@@ -135,80 +135,62 @@ class WsEndpoint extends EventEmitter {
 
     sendSync(recipientId, message) {
         const recipientAddress = this.resolveAddress(recipientId)
-        if (!this.isConnected(recipientAddress)) {
+        if (!this.isConnected(recipientAddress) || !this.connections.has(recipientAddress)) {
```
Comment (harbu) [inline src/connection/WsEndpoint.js]: But isn't `this.isConnected()` already effectively `this.connections.has`?

File: src/connection/WsEndpoint.js [hunk 2]
```diff
@@ -135,80 +135,62 @@ class WsEndpoint extends EventEmitter {
...
+                this._socketSend(ws, message, recipientId, recipientAddress)
+            })
         }
     }
 
     send(recipientId, message) {
         const recipientAddress = this.resolveAddress(recipientId)
         return new Promise((resolve, reject) => {
-            if (!this.isConnected(recipientAddress)) {
+            if (!this.isConnected(recipientAddress) || !this.connections.has(recipientAddress)) {
```
Comment (harbu) [inline src/connection/WsEndpoint.js]: Same as above

**Review by harbu — CHANGES_REQUESTED** [review]
Nice! See comments, I think you don't need new conditions in the if statements?

File: src/connection/WsEndpoint.js (hunk 1 above)
Comment (mirpo) [reply src/connection/WsEndpoint.js]: Good point!

**Review by harbu — APPROVED** [review]

**Pull request merged**